receiver/dockerstatsreceiver/receiver.go

```diff
@@ -45,13 +46,32 @@ type metricsReceiver struct {
 }
 
 func newMetricsReceiver(set receiver.Settings, config *Config) *metricsReceiver {
+	// Resolve the Docker endpoint during receiver initialization
+	config.Endpoint = config.resolveDockerEndpoint()
+
 	return &metricsReceiver{
 		config:   config,
 		settings: set,
 		mb:       metadata.NewMetricsBuilder(config.MetricsBuilderConfig, set),
 	}
 }
 
+const (
+	defaultEndpoint = "unix:///var/run/docker.sock"
+	dockerHostEnv   = "DOCKER_HOST"
+)
+
+func (c *Config) resolveDockerEndpoint() string {
+	if c.Endpoint != "" {
+		return c.Endpoint
+	}
+
+	if dockerHost := os.Getenv(dockerHostEnv); dockerHost != "" {
+		return dockerHost
+	}
+	return defaultEndpoint
+}
+
 func (r *metricsReceiver) start(ctx context.Context, _ component.Host) error {
 	var err error
 	r.client, err = docker.NewDockerClient(&r.config.Config, r.settings.Logger)
```

Comment:
So annoyingly, this is all you need to do here:

```suggestion
	var opts []client.Opt
	if r.config.Config.Endpoint == "" {
		opts = append(opts, client.WithHostFromEnv())
	}

	var err error
	r.client, err = docker.NewDockerClient(&r.config.Config, r.settings.Logger, opts...)
```

Reply:
I would also suggest removing the default value in the new config method since it will now just respect the environment settings.

Reply:
Thank you for suggesting this. I have done the needful